src/runtime/CollectionWrappers/ListWrapper.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Collections.Generic;
+
+namespace Python.Runtime.CollectionWrappers
+{
+    internal class ListWrapper<T> : SequenceWrapper<T>, IList<T>
+    {
+        public ListWrapper(PyObject pyObj) : base(pyObj)
+        {
+
+        }
+
+        public T this[int index]
+        {
+            get
+            {
+                var item = Runtime.PyList_GetItem(pyObject.Handle, index);
+                var pyItem = new PyObject(item);
```

Comment:
Dispose this

Reply:
@lostmsu again I thought the point of making it IDisposable was that the finalizer would take care of that

Reply:
Wrap it by using statement is better. If you knew it's no use anymore you should dispose it manually instead of waiting for the auto collection to do it.

Reply:
@lostmsu @amos402 sorry, I might be missing context (I rebased and I don't know if this comment lines up with the code it was pointing to initially).

What specifically should I be disposing or wrap in a using statement?

Reply:
```c#
using (var pyItem = new PyObject(item)) { }
```

Reply:
@koubaa this still needs a fix. With C# 8+ which we use now, it should just be `using var pyItem = new PyObject(item);` so that the Python object, corresponding to the `item` would be disposed and not pollute finalizer thread unless duplicated.